### DISCUSSION TITLE
Where to keep HOMEBREW_GITHUB_API_TOKEN other than a dot file.

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#2720] [Everyday usage] [answered]
Looking for advice...

I understand this is generally a System Environment topic, but thinking that maybe Homebrew can access the API Token from another location?

For those of us that keep our dot files on GitHub and don't want to keep passwords and API Tokens in them for that reason. What is an alternative way to make Homebrew aware of this Token?

**Comment by neutric** [comment, marked as answer]
Depending on your needs, putting the token in a file that doesn't leave your computer and sourcing that file from your shell's dotfile might do the trick. Or search for "dotfiles secrets" upon which you find e.g. https://github.com/twpayne/chezmoi. Feel free to ask further questions. If satisfied, please mark this as answered.

**Comment by alebcay** [comment]
In addition to the answer above, Homebrew will try and use credentials in the macOS keychain if present. You can use the `osxkeychain` credential helper with your GitHub username as the username and your token as the password.
